Add documentation for slab benchmark multiphysics problem.

meltawila opened the pull request:
Closes #98

meltawila committed:
rebased on top of upstream/devel

meltawila committed:
formatting and adding more docs

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/3885091/), step Sync to remote on 084b609 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/salamander/docs/PRs/99/site)

This comment will be updated on new commits.

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/3885089/), step Generate coverage on 084b609 wanted to post the following:


## Coverage

Coverage did not change

[Full coverage report](https://mooseframework.inl.gov/salamander/docs/PRs/99/coverage)



This comment will be updated on new commits.

meltawila committed:
rebased on upstream/devel

meltawila commented:
@simopier would you activate the tests please?

simopier commented on doc/content/bib/references.bib:
```diff
@@ -38,3 +38,21 @@ @book{lieberman1994principles
+  organization = {ANS},
+  year         = {2022},
+  month        = {07},
+  booktitle = {Proceedings of  PHYSOR},
```
```suggestion
  booktitle = {Proceedings of PHYSOR},
```

meltawila replied:
Added in my commit

simopier commented on doc/content/verification_validation_examples/slab_benchmark/model/make_openmc_model_dagmc.py:
```diff
@@ -0,0 +1,95 @@
+Tmax = 393
+Tmin = 293
+L0 = 100
+L = 106.47
```
```suggestion
T0 = 293 # K
Tmax = 393 # K
Tmin = 293 # K
L0 = 100 # cm
L = 106.47 # cm
```

meltawila replied:
Added in my commit

simopier commented on test/tests/neutronics/expanding_slab/model.py:
```diff
@@ -0,0 +1,95 @@
+import openmc
```
Please add a comment describing ht this file does.

meltawila replied:
Added in my commit

simopier commented on test/tests/neutronics/expanding_slab/openmc.i:
```diff
@@ -0,0 +1,175 @@
+P = 1.0e22 # eV/s
```
Please add a comment explaining what this file does.

meltawila replied:
Added in my commit

simopier commented on doc/content/verification_validation_examples/slab_benchmark/model/openmc.i:
```diff
@@ -0,0 +1,175 @@
+P = 1.0e22 # eV/s
+eV_to_J = 1.602e-19 # J/eV
```
Use the value from PhysicalConstants.h instead.

meltawila replied:
Cardinal doesn't use the units system, so this won't work

simopier replied:
See new comment.

simopier commented on doc/content/verification_validation_examples/slab_benchmark/model/openmc.i:
```diff
@@ -0,0 +1,175 @@
+[ICs]
+  [temp]
+    type = FunctionIC
+    variable = temp
```
```suggestion
  [temperature]
    type = FunctionIC
    variable = temperature
```

simopier commented on doc/content/verification_validation_examples/slab_benchmark/model/openmc.i:
```diff
@@ -0,0 +1,175 @@
+  [temp]
+    type = FunctionIC
+    variable = temp
+    function = 293
```
```suggestion
    function = 293 # K
```

simopier replied:
missed suggestion.

simopier commented on doc/content/verification_validation_examples/slab_benchmark/model/openmc.i:
```diff
@@ -0,0 +1,175 @@
+    displaced_source_mesh = true
+    displaced_target_mesh = true
+  []
+  [temp_from_solid]
```
```suggestion
  [temperature_from_solid]
```

simopier commented on doc/content/verification_validation_examples/slab_benchmark/model/openmc.i:
```diff
@@ -0,0 +1,175 @@
+    type = MultiAppGeneralFieldNearestLocationTransfer
+    from_multi_app = solid
+    variable = temp
+    source_variable = temp_praux
```
```suggestion
    variable = temperature
    source_variable = temperature_praux
```

simopier commented on doc/content/verification_validation_examples/slab_benchmark/model/openmc.i:
```diff
@@ -0,0 +1,175 @@
+[VectorPostprocessors]
+  [temp]
+    type = ElementValueSampler
+    variable = 'temp'
```
```suggestion
  [temperature]
    type = ElementValueSampler
    variable = 'temperature'
```

simopier commented on doc/content/verification_validation_examples/slab_benchmark/model/openmc.i:
```diff
@@ -0,0 +1,175 @@
+    type = MoabSkinner
+    execute_on = 'timestep_begin'
+    verbose = true
+    temperature = temp
```
```suggestion
    temperature = temperature
```

simopier commented on test/tests/neutronics/expanding_slab/solid.i:
```diff
@@ -0,0 +1,217 @@
+T0 = 293 # K
```
Please add a comment describing what this file does.

meltawila replied:
Added in my commit

simopier commented on test/tests/neutronics/expanding_slab/solid.i:
```diff
@@ -0,0 +1,217 @@
+q = 1e8 # eV
+k0= 1.25e19 # eV/(s-cm-K^2) k(T) = k0 T(x)
+phi0 = 2.5e14 # 1/s-cm^2 (flux at the origin)
+eV_to_J = 1.602e-19 # J/eV
```
use value from PhysicalConstants.h

simopier commented on doc/content/verification_validation_examples/slab_benchmark/model/solid.i:
```diff
@@ -0,0 +1,217 @@
+[]
+
+[Variables]
+  [temp]
```
```suggestion
  [temperature]
```

simopier commented on doc/content/verification_validation_examples/slab_benchmark/model/solid.i:
```diff
@@ -0,0 +1,217 @@
+        add_variables = true
+        eigenstrain_names = eigenstrain
+        generate_output = 'strain_xx'
+        temperature = temp
```
```suggestion
        temperature = temperature
```

simopier commented on doc/content/verification_validation_examples/slab_benchmark/model/solid.i:
```diff
@@ -0,0 +1,217 @@
+[ICs]
+  [temp]
+    type = FunctionIC
+    variable = temp
```
```suggestion
  [temperature]
    type = FunctionIC
    variable = temperature
```

simopier commented on doc/content/verification_validation_examples/slab_benchmark/model/solid.i:
```diff
@@ -0,0 +1,217 @@
+  []
+  [dt]
+    type = HeatConductionTimeDerivative
+    variable = temp
```
```suggestion
    variable = temperature
  []
  [kappa_fission]
    type = CoupledForce
    variable = temperature
    v = kappa_fission
  []
  [dt]
    type = HeatConductionTimeDerivative
    variable = temperature
```

simopier commented on doc/content/verification_validation_examples/slab_benchmark/model/solid.i:
```diff
@@ -0,0 +1,217 @@
+   [temp_projection]
+     type = ProjectionAux
+     variable = temp_praux
+     v = temp
```
```suggestion
     variable = temperature_praux
     v = temperature
```

simopier commented on doc/content/verification_validation_examples/slab_benchmark/model/solid.i:
```diff
@@ -0,0 +1,217 @@
+[Materials]
+  [thermal_parameters]
+    type = HeatConductionMaterial
+    temp = temp
```
```suggestion
    temp = temperature
```

simopier commented on test/tests/neutronics/expanding_slab/solid.i:
```diff
@@ -0,0 +1,217 @@
+  [elasticity_tensor]
+    type = ComputeIsotropicElasticityTensor
+    youngs_modulus = 1e6
+    poissons_ratio = 0.0
```
I suggest adding these numbers at the top of the file and provide units for them.

meltawila replied:
The constants at the top are taken directly from the benchmarks, while these are placeholder values. We don't have stresses to compare against, and c_p and density were added to add a time derivative in the heat transfer eqn, but we're only interested in the final steady state solution that the benchmark solves. Any non-extreme values here should give the same results

simopier commented on doc/content/verification_validation_examples/slab_benchmark/model/solid.i:
```diff
@@ -0,0 +1,217 @@
+    stress_free_temperature = ${T0}
+    thermal_expansion_function_reference_temperature = ${T0}
+    thermal_expansion_function = cte_func_mean
+    temperature = temp
```
```suggestion
    temperature = temperature
```

simopier commented on doc/content/verification_validation_examples/slab_benchmark/model/solid.i:
```diff
@@ -0,0 +1,217 @@
+  [sides_convective_BC]
+      type = ConvectiveFluxFunction
+      T_infinity = ${T0}
+      variable = temp
```
```suggestion
      variable = temperature
```

simopier commented on test/tests/neutronics/expanding_slab/solid.i:
```diff
@@ -0,0 +1,217 @@
+  type = Transient
+  solve_type = 'PJFNK'
+  verbose = true
+  nl_abs_tol = 5e-6
```
This value is very high. We usually use something much smaller (1e-9).

simopier commented on doc/content/verification_validation_examples/slab_benchmark/model/solid.i:
```diff
@@ -0,0 +1,217 @@
+  type = ReferenceResidualProblem
+  extra_tag_vectors = 'ref'
+  reference_vector = 'ref'
+[]
```
This block usually goes next to the mesh block. Not a big deal at all, just something to keep in mind for the future.

simopier commented on doc/content/bib/articles_using_salamander.bib:
```diff
@@ -8,3 +8,10 @@ @article{Franklin2025
+
+@article{eltawila2025,
+    title = {Thermomechanics Coupling to Monte Carlo Particle Transport on Unstructured Mesh Geometries},
+    year = {2025},
```
```suggestion
    year = {2025},
    journal = {Journal TBD, Preprint on \url{https://ssrn.com/abstract=5379499}},
```
make sure to build the documentation locally before pushing to fix these issues. 
so that the documentation failure stops.

aprilnovak replied:
update the citation to the accepted title

simopier commented on doc/content/verification_validation_examples/slab_benchmark/slab_benchmark.md:
```diff
@@ -0,0 +1,291 @@
+We use total mass preservation to get material density change from the initial to the final state, starting from:
+
+\begin{equation}
+  \rho(x)\ell(x) = \rho_0\ell_0. \label{eqn:mass_conservation}
```
```suggestion
  \rho(x)\ell(x) = \rho_0\ell_0, \label{eqn:mass_conservation}
```

simopier commented on doc/content/verification_validation_examples/slab_benchmark/slab_benchmark.md:
```diff
@@ -0,0 +1,291 @@
+  \rho(x)\ell(x) = \rho_0\ell_0. \label{eqn:mass_conservation}
+\end{equation}
+
+To get the relation
```
```suggestion
to get the relation
```

simopier commented on doc/content/verification_validation_examples/slab_benchmark/slab_benchmark.md:
```diff
@@ -0,0 +1,291 @@
+
+## Analytic solutions
+
+the reader is referred to the original benchmark publication [!citep](griesheimer2022analytical) for full details on the derivation of these solution. Here we present the final formulas that we use later for comparison with our models.
```
```suggestion
The reader is referred to the original benchmark publication [!citep](griesheimer2022analytical) for full details on the derivation of these solutions. Here we present the final formulas that we use later for comparison with our models.
```

simopier commented on doc/content/verification_validation_examples/slab_benchmark/slab_benchmark.md:
```diff
@@ -0,0 +1,291 @@
+The example set of model parameter values shown in [tab:params] are the suggested benchmark parameters [!citep](griesheimer2022analytical), resulting in $k_\text{eff} = 0.29557$ and a heated slab length of $L = 106.47$ cm with $P = 1.0\times10^{22}$ eV/s.
+
+!table id=tab:params caption=Example set of model parameters with two different power scenarios [!citep](griesheimer2022analytical).
+| **Parameter                  | **Value                   | **Unit                              |
```
Why do you use the `**` symbols everywhere? What do they do? 
I also don't see the need for the ` signs around the units.

simopier commented on doc/content/verification_validation_examples/slab_benchmark/slab_benchmark.md:
```diff
@@ -0,0 +1,291 @@
+| **$A$**                        | 180                         | `[g/mol`$^{-1}$`]`                        |
+| **$T_0$**                      | 293                         | `[K]`                                   |
+| **$q$**                        | 1.0E8                       | `[eV]`                                  |
+| **$\kappa_0$**                 | 1.25E19.                    | `[eV/s/cm/K`$^{-2}$`]`      |
```
```suggestion
| **$\kappa_0$**                 | 1.25 $\times$ 10$^{19}$.                    | `[eV/s/cm/K`$^{-2}$`]`      |
```

simopier commented on doc/content/verification_validation_examples/slab_benchmark/slab_benchmark.md:
```diff
@@ -0,0 +1,291 @@
+| **$L_0$**                      | 100                         | `[cm]`                                  |
+| **$A$**                        | 180                         | `[g/mol`$^{-1}$`]`                        |
+| **$T_0$**                      | 293                         | `[K]`                                   |
+| **$q$**                        | 1.0E8                       | `[eV]`                                  |
```
```suggestion
| **$q$**                        | 1.0 $\times$ 10$^{8}$           | `[eV]`                                  |
```

simopier commented on doc/content/verification_validation_examples/slab_benchmark/slab_benchmark.md:
```diff
@@ -0,0 +1,291 @@
+| **$T_0$**                      | 293                         | `[K]`                                   |
+| **$q$**                        | 1.0E8                       | `[eV]`                                  |
+| **$\kappa_0$**                 | 1.25E19.                    | `[eV/s/cm/K`$^{-2}$`]`      |
+| **$\phi_0$**                   | 2.5E14                      | `[#/cm`$^2$`/s]`                  |
```
```suggestion
| **$\phi_0$**                   | 2.5 $\times$ 10$^{24}$        | `[#/cm`$^2$`/s]`                  |
```

simopier commented on doc/content/verification_validation_examples/slab_benchmark/slab_benchmark.md:
```diff
@@ -0,0 +1,291 @@
+| **$\phi_0$**                   | 2.5E14                      | `[#/cm`$^2$`/s]`                  |
+| **$\nu\Sigma_f/\Sigma_t$**     | 1.5                         | `[-]`                                   |
+| **$\Sigma_s/\Sigma_t$**        | 0.45                        | `[-]`                                   |
+| **$P$**                        | 1.0E22                      | `[eV/s]`                         |
```
```suggestion
| **$P$**                        | 1.0 $\times$ 10$^{22}$        | `[eV/s]`                         |
```

simopier commented on doc/content/verification_validation_examples/slab_benchmark/slab_benchmark.md:
```diff
@@ -0,0 +1,291 @@
+This model is using S$_2$ neutron transport to compare with the 1D Analytic benchmark. To replicate
+the results, you have build your application with [this](https://github.com/meltawila/openmc_S2transport) patch of OpenMC.
+
+Cardinal is used to couple OpenMC with MOOSE via Picard iteration. OpenMC tallies neutron flux, heating, and the multiplication factor. A conceptual depiction of the meshes used as well as the data transfers that occur on each Picard iteration is shown in [fig:schematic]. Three different meshes/geometries are involved. The thermomechanics mesh is a simple 1D mesh of $N$ EDGE2 elements; the thermo ...
```
```suggestion
Cardinal is used to couple OpenMC with MOOSE via Picard iteration. OpenMC tallies neutron flux, heating, and the multiplication factor. A conceptual depiction of the meshes used as well as the data transfers that occur on each Picard iteration is shown in [fig:schematic]. 
Three different meshes/geometries are involved. The thermomechanics mesh is a simple 1D mesh of $N$ EDGE2 elements; the thermomechanical physics do not require higher dimensional (e.g., 2D or 3D) meshes for this problem. The OpenMC geometry is represented using DAGMC, where cells are bounded by a triangulated surface (TRI3 elements). 
To facilitate data transfers between OpenMC and MOOSE thermomechanics, there is also an intermediate volumetric data transfer mesh composed of TET4 elements; no physics solves happen on this intermediate mesh. 
[fig:schematic] highlights one element (for MOOSE and the Cardinal intermediate data transfer mesh) and one OpenMC cell each in blue, with adjacent portions of the geometry shown in gray for context.
```

simopier commented on doc/content/verification_validation_examples/slab_benchmark/slab_benchmark.md:
```diff
@@ -0,0 +1,291 @@
+
+## Results
+
+We see the heating result in [fig:heatingsol] compared against the analytic solution, as well as flux and temperature solutions in [fig:fluxsol] and [fig:tempsol]. Figures are only included for P=1.0E22 `[eV/s]` and on a fine mesh where the number of OpenMC and mesh elements is $N = 100$, since all results have a similar shape. Both the heating and flux tally everywhere agrees with the analytic s ...
```
```suggestion
We see the heating result in [fig:heatingsol] compared against the analytic solution, as well as flux and temperature solutions in [fig:fluxsol] and [fig:tempsol]. Figures are only included for $P=1.0 \times 10^{22}$ eV/s and on a fine mesh where the number of OpenMC and mesh elements is $N = 100$, since all results have a similar shape. Both the heating and flux tally everywhere agrees with the analytic solution within 3$\sigma$.
```

simopier commented on doc/content/verification_validation_examples/slab_benchmark/slab_benchmark.md:
```diff
@@ -0,0 +1,291 @@
+  caption=Temperature result from MOOSE for N = 100 mesh elements compared against the analytic solution.
+  style=display:block;margin-left:auto;margin-right:auto;width:40%;
+
+Table [tab:results] shows results convergence with mesh refinement.
```
```suggestion
[tab:results] shows results convergence with mesh refinement.
```
You shouldn't need `Table` here.

simopier commented on doc/content/verification_validation_examples/slab_benchmark/slab_benchmark.md:
```diff
@@ -0,0 +1,291 @@
+Table [tab:results] shows results convergence with mesh refinement.
+
+!table id=tab:results caption=Results and corresponding errors with different mesh sizes [!citep](eltawila2025).
+| **Resolution        | **Heated length `[cm]` | **Error `[μm]` | **$k_{\text{eff}}$** | **Error `[pcm]` |
```
Same comment about ** and `

simopier commented on doc/content/verification_validation_examples/slab_benchmark/slab_benchmark.md:
```diff
@@ -0,0 +1,291 @@
+| **Resolution        | **Heated length `[cm]` | **Error `[μm]` | **$k_{\text{eff}}$** | **Error `[pcm]` |
+| :- | :- | :- | :- | :- |
+| **Analytic solution | 106.47                 | --             | 0.29557              | --              |
+| **5                 | 106.3298               | -1402          | 0.29608 ± 0.00001    | 51 ± 1          |
```
Once the documentation builds, I'll need to check that the plus/minus signs work in markdown.

simopier commented on doc/content/verification_validation_examples/slab_benchmark/slab_benchmark.md:
```diff
@@ -0,0 +1,291 @@
+| **100               | 106.4745               | 45             | 0.29549 ± 0.00001    | -8 ± 1          |
+| **200               | 106.4748               | 48             | 0.29556 ± 0.00001    | -1 ± 1          |
+
+In [fig:fluxconv] and [fig:tempconv] convergence of flux and temperature relative L$^2$ norms, defined as
```
```suggestion
In [fig:fluxconv] and [fig:tempconv], convergence of flux and temperature relative L$^2$ norms, defined as
```

simopier reviewed (changes requested):
This looks great overall!
However, I would recommend adding a test to run a very small version of this as part of the testing suite to make sure we maintain this capability available to users.

meltawila committed:
update docs and address review comments

meltawila committed:
adding tests

simopier commented on doc/content/bib/articles_using_salamander.bib:
```diff
@@ -11,7 +11,8 @@ @article{Franklin2025
 
 @article{eltawila2025,
     title = {Thermomechanics Coupling to Monte Carlo Particle Transport on Unstructured Mesh Geometries},
+    journal = {SSRN Preprint},
```
To be updated

simopier commented on doc/content/verification_validation_examples/slab_benchmark/model/openmc.i:
```diff
@@ -0,0 +1,177 @@
+# Cardinal input file for slab benchmark OpenMC model to couple with thermomechanics
+
+P = 1.0e22 # eV/s
+eV_to_J = 1.602e-19 # J/eV
```
```suggestion
eV_to_J = 1.602176634e-19 # J/eV
```
This is what I meant by `Use the value from PhysicalConstants.h instead.`

simopier commented on test/tests/neutronics/expanding_slab/solid.i:
```diff
@@ -0,0 +1,219 @@
+q = 1e8 # eV
+k0= 1.25e19 # eV/(s-cm-K^2) k(T) = k0 T(x)
+phi0 = 2.5e14 # 1/s-cm^2 (flux at the origin)
+eV_to_J = 1.602e-19 # J/eV
```
```suggestion
eV_to_J = 1.602176634e-19 # J/eV
```

simopier commented on test/tests/neutronics/moving_mesh/cell_tally.i:
```diff
@@ -0,0 +1,132 @@
+[GlobalParams]
```
Add a comment at the top of the input file, just like I requested for the others.

simopier commented on test/tests/neutronics/moving_mesh/mesh_tally.i:
```diff
@@ -0,0 +1,133 @@
+[GlobalParams]
```
Add a comment at the top of the input to describe what it does.

simopier replied:
It seems like this input file has a lot of similarities with the call_tally.i input file. Do you think it would make sense to generate a base input file and utilize the `!include` capabilities to reduce duplication?

simopier commented on doc/content/verification_validation_examples/slab_benchmark/slab_benchmark.md:
```diff
@@ -181,21 +181,21 @@ Substituting [eqn:flux_solution] and [eqn:temperature] into [eqn:constraint] giv
 \end{equation}
 
-The example set of model parameter values shown in [tab:params] are the suggested benchmark parameters [!citep](griesheimer2022analytical), resulting in $k_\text{eff} = 0.29557$ and a heated slab length of $L = 106.47$ cm with $P = 1.0\times10^{22}$ eV/s.
+The example set of model parameter values shown in [tab:params] are the suggested benchmark parameters [!citep](griesheimer2022analytical), resulting in $k_\text{eff} = 0.29557$ and a heated slab length of $L = 106.47$ `cm` with $P = 1.0\times10^{22}$ `eV/s`.
```
General comment: I do not think you need the "`" signs around units. We usually do not use them in papers and in the rest of the documentation. For consistency, I recommend removing them.

simopier reviewed (changes requested):
Thank you @meltawila!

Some additional comments on the new commits. 
Things are moving in the right directlion. 

By the way, for future rounds and PRs, it makes it easier over rounds of reviews if you incorporate the suggestions directly in the `Files changed` tab on the PR page. It incorporates the exact suggestion and then resolves the comment.

simopier commented on doc/content/verification_validation_examples/slab_benchmark/slab_benchmark.md:
```diff
@@ -0,0 +1,291 @@
+
+This verification problem presents capabilities to couple neutron transport with thermal conduction and material expansion. The analytic benchmark slab problem [!citep](griesheimer2022analytical) describes a slab where S$_2$ neutron transport is coupled with thermal conduction, convective boundary conditions, Doppler-broadened cross sections, and nonlinear thermal expansion effects along the heat ...
+
+While the restriction to S$_2$ neutron transport would be considered overly simplistic for most radiation transport  simulations, the objective here is to verify thermomechanical-OpenMC coupling using analytic solutions for three coupled solution fields (Displacements, temperature, and flux). Simplifications of each individual physics included in this model are not imposed by inherent limitations ...
```
```suggestion
While the restriction to S$_2$ neutron transport would be considered overly simplistic for most radiation transport  simulations, the objective here is to verify thermomechanical-OpenMC coupling using analytic solutions for three coupled solution fields (displacements, temperature, and flux). Simplifications of each individual physics included in this model are not imposed by inherent limitations of the moving-mesh capabilities and only aim to allow the study on the interactions between multiphysics with an analytic solution.
```

simopier commented on doc/content/verification_validation_examples/slab_benchmark/slab_benchmark.md:
```diff
@@ -0,0 +1,291 @@
+
+\begin{equation}
+  \label{eqn:rho_t}
+  \rho(x) = \rho_0\sqrt{\frac{T_0}{T(x)}}
```
```suggestion
  \rho(x) = \rho_0\sqrt{\frac{T_0}{T(x)}}.
```

simopier commented on doc/content/verification_validation_examples/slab_benchmark/slab_benchmark.md:
```diff
@@ -0,0 +1,291 @@
+
+## Analytic solutions
+
+The reader is referred to the original benchmark publication [!citep](griesheimer2022analytical) for full details on the derivation of these solutions. Here we present the final formulas that we use later for comparison with our models.
```
```suggestion
The reader is referred to the original benchmark publication [!citep](griesheimer2022analytical) and the updates made in [!citep](eltawila2025) for full details on the derivation of these solutions. Here we present the final formulas that we use later for comparison with our models.
```

simopier commented on doc/content/verification_validation_examples/slab_benchmark/slab_benchmark.md:
```diff
@@ -0,0 +1,291 @@
+
+The example set of model parameter values shown in [tab:params] are the suggested benchmark parameters [!citep](griesheimer2022analytical), resulting in $k_\text{eff} = 0.29557$ and a heated slab length of $L = 106.47$ `cm` with $P = 1.0\times10^{22}$ `eV/s`.
+
+!table id=tab:params caption=Example set of model parameters with two different power scenarios [!citep](griesheimer2022analytical).
```
You describe two power scenarios here, but you only list one in the table. Make sure to be consistent.

simopier commented on doc/content/verification_validation_examples/slab_benchmark/slab_benchmark.md:
```diff
@@ -0,0 +1,291 @@
+This model is using S$_2$ neutron transport to compare with the 1D Analytic benchmark. To replicate
+the results, you have build your application with [this](https://github.com/meltawila/openmc_S2transport) patch of OpenMC.
+
+Cardinal is used to couple OpenMC with MOOSE via Picard iteration. OpenMC tallies neutron flux, heating, and the multiplication factor. A conceptual depiction of the meshes used as well as the data transfers that occur on each Picard iteration is shown in [fig:schematic]. Three different meshes/geometries are involved. The thermomechanics mesh is a simple 1D mesh of $N$ EDGE2 elements; the thermo ...
```
```suggestion
Cardinal is used to couple OpenMC with MOOSE via Picard iteration. OpenMC tallies neutron flux, heating, and the multiplication factor. A conceptual depiction of the meshes used as well as the data transfers that occur on each Picard iteration is shown in [fig:schematic]. Three different meshes/geometries are involved. The thermomechanics mesh is a simple 1D mesh of $N$ EDGE2 elements; the thermomechanical physics do not require higher dimensional (e.g., 2D or 3D) meshes for this problem. The OpenMC geometry is represented using DAGMC, where cells are bounded by a triangulated surface (TRI3 elements). To facilitate data transfers between OpenMC and MOOSE thermomechanics, there is also an intermediate volumetric data transfer mesh composed of TET4 elements; no physics solves happen on this intermediate mesh. [fig:schematic] highlights one element (for MOOSE and the Cardinal intermediate data transfer mesh) and one OpenMC cell each in blue, with adjacent portions of the geometry shown in gray for context.
```

simopier commented on test/tests/neutronics/moving_mesh/tests:
The tests here do not reprocude the results from the paper, and are testing slightly other things using other input files. 
I therefore have several reauests:
- could you please add a section in the documentation to explain these tests and describe the input files being used here?
- I'm not a fan of the fact that the input files relevant to the paper are not tested and will therefore likely get out of date as SALAMANDER/CARDINAL/OpenMC/MOOSE continue to evolve. If we can't reprocude the paper results with the regular build, is it not possible to create a test reciepe to use the OpenMC patch you need for this? At the very least, you should have all the input files go through a syntax check in one of the tests, or run for a short amount of time just to make sure they at least run. This should all be detailed a bit on the documentation page.

simopier reviewed (changes requested):
Some more geenral comments on the tests and documentation now that it builds. 
Thank you!

simopier commented:
Do you need anything to finalize this, @meltawila?

meltawila committed:
addressing review comments

meltawila committed:
removing redundant tests - now in /expanding_slab

meltawila committed:
updating test files

meltawila committed:
formatting (Ref. #98)

meltawila committed:
update listings path

meltawila commented:
@simopier I addressed your comments and updated the tests. This still won't build on Civet until Cardinal is updated, but if you'd like you can use my patch with
```
cd cardinal
git remote add mahmoud https://github.com/meltawila/cardinal.git
git fetch mahmoud
git cherry-pick ff75e9616bfd68ed76e9d15c693d936d4ac84276
```
I originally added the slab benchmark model under '/doc' and had a derivative of it in '/tests' since this case needs to use the openmc_s2patch to result the correct solutions. But to avoid this duplication and since you also expressed concern about these above, I removed the '/moving_mesh' tests and moved '/expanding_slab' to the test directory and added an alert in slab_benchmark.md with instructions on how to build SALAMANDER with the s2patch to get the benchmark results correctly. The removed tests were not intended to test anything new other than what is already in '/expanding_slab'.

Otherwise I think I addressed everything

meltawila commented:
Hi @simopier, would you activate the tests here now that the Cardinal submodule is updated?

simopier commented:
> Hi @simopier, would you activate the tests here now that the Cardinal submodule is updated?

You *might* have to re-push to get everything up to date.

meltawila committed:
updated max_parallel and docs

meltawila commented:
`particles=1000` might be too little - I'll try one more time with max_parallel=1. if it failed again I'd increase the number of particles for this test

meltawila committed:
trying again with max_parallel=1

meltawila committed:
update test openmc parameters

meltawila committed:
fix formatting in slab_benchmark.md

meltawila commented:
@simopier would you activate the tests here please?

simopier reviewed (changes requested):
Some additional comments and suggestions before this can be merged.

meltawila committed:
Apply suggestions from @simopier review

Co-authored-by: Pierre-Clement Simon <pierreclement.simon@gmail.com>

simopier reviewed (commented):
Suggestions were applied. Broader review comments still need to be addressed.

meltawila committed:
tightened tolerances, added constants at the top of the file, and regolded

meltawila commented:
@simopier yep I was just rerunning the full case to make sure it works fine with the new edits. I added these now. Cardinal test suite uses `max_parallel` regularly as we decrease the number of particles significantly to decrease run time and OpenMC will error if there aren't enough source particles in the fission bank on a process. I have a comment in tests file to say that

moosebuild commented:
Job [Build test summary](https://civet.inl.gov/job/3885088/), step Build test summary on 084b609 wanted to post the following:

## Test summary

Compared against 201793a in job [civet.inl.gov/job/3893681](https://civet.inl.gov/job/3893681).

### Added tests

| Test                                                  | Time (s)   | Memory (MB)   |
|-------------------------------------------------------|------------|---------------|
| `test:neutronics/expanding_slab.slab_thermomechanics` | SKIP       | 0.00          |
| `test:neutronics/expanding_slab.run_snmodel`          | SKIP       | 0.00          |

simopier reviewed (approved):
Looks great to me, thank you @meltawila!
